addons/account/models/account_account.py

```diff
@@ -597,37 +614,34 @@ def _compute_opening_debit_credit(self):
     @api.depends('code')
     def _compute_account_type(self):
         accounts_to_process = self.filtered(lambda account: account.code and not account.account_type)
-        self._get_closest_parent_account(accounts_to_process, 'account_type', default_value='asset_current')
+
+        if accounts_to_process:
+            company_accounts = self.search(self._check_company_domain(self.env.company))
+
+            for account in accounts_to_process:
+                closest_parent_account = self._get_closest_parent_account(account.code, company_accounts)
+                account.account_type = closest_parent_account.account_type if closest_parent_account else 'asset_current'
 
     @api.depends('code')
     def _compute_account_tags(self):
         accounts_to_process = self.filtered(lambda account: account.code and not account.tag_ids)
-        self._get_closest_parent_account(accounts_to_process, 'tag_ids', default_value=[])
 
-    def _get_closest_parent_account(self, accounts_to_process, field_name, default_value):
-        """
-            This helper function retrieves the closest parent account based on account codes
-            for the given accounts to process and assigns the value of the parent to the specified field.
+        if accounts_to_process:
+            company_accounts = self.search(self._check_company_domain(self.env.company))
 
-            :param accounts_to_process: Records of accounts to be processed.
-            :param field_name: Name of the field to be updated with the closest parent account value.
-            :param default_value: Default value to be assigned if no parent account is found.
-        """
-        assert field_name in self._fields
+            for account in accounts_to_process:
+                closest_parent_account = self._get_closest_parent_account(account.code, company_accounts)
+                account.tag_ids = closest_parent_account.tag_ids if closest_parent_account else []
 
-        all_accounts = self.search_read(
-            domain=self._check_company_domain(self.env.company),
-            fields=['code', field_name],
-            order='code',
-        )
-        accounts_with_codes = {}
-        # We want to group accounts by company to only search for account codes of the current company
-        for account in all_accounts:
-            accounts_with_codes[account['code']] = account[field_name]
-        for account in accounts_to_process:
-            codes_list = list(accounts_with_codes.keys())
-            closest_index = bisect_left(codes_list, account.code) - 1
-            account[field_name] = accounts_with_codes[codes_list[closest_index]] if closest_index != -1 else default_value
+    def _get_closest_parent_account(self, code, accounts_to_consider):
```

Comment:
The name of the function is horrible indeed, but you are adding 20 loc of duplicated code :/
Not sure it is better, so I'd like to avoid the diff to avoid useless friction (conflicts, forward ports, API change in stable,...)